File: MyToast/app/src/main/java/com/example/toast/MainActivity.java
package com.example.toast;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //토스트 위치 정해주기
        Button button =(Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Toast toast = Toast.makeText(getApplicationContext(), "위치가 바뀐 토스트 ", Toast.LENGTH_LONG);
                toast.setGravity(Gravity.TOP|Gravity.LEFT,200,200);
                toast.show();
            }
            });

        //토스트 모양 바꾸기
        Button button2 =(Button) findViewById(R.id.button2);
        button.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                LayoutInflater inflater = getLayoutInflater();
                View layout=inflater.inflate(R.layout.toastborder,(ViewGroup)findViewById(R.id.toast_layout_root));

                TextView text = (TextView)layout.findViewById(R.id.text);
                text.setText("모양을 바꾼 toast입니다");

                //토스트 객체 생성
                Toast toast = new Toast(getApplicationContext());
                toast.setGravity(Gravity.CENTER,0,-100);
                toast.setDuration(Toast.LENGTH_LONG);
                toast.setView(layout);

                toast.show();
            }
        });
    }
}

File: Kotlin/Retrofit/app/src/main/java/com/example/retrofit/MainActivity.kt
package com.example.retrofit

import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.retrofit.utils.Constans.TAG
import com.example.retrofit.utils.SEARCH_TYPE
import com.example.retrofit.utils.onMyTextChanged
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.layout_button_search.*

class MainActivity : AppCompatActivity() {

    private var currentSearchType: SEARCH_TYPE = SEARCH_TYPE.PHOTO

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        search_term_radio_group.setOnCheckedChangeListener { _, checkedId ->
            when (checkedId) {
                R.id.photo_search_radio_btn -> {
                    Log.d(TAG, "사진 버튼 클릭!")
                    search_term_text_layout.hint = "사진검색"
                    search_term_text_layout.startIconDrawable =
                        resources.getDrawable(
                            R.drawable.ic_baseline_photo_library_24,
                            resources.newTheme()
                        )
                    this.currentSearchType = SEARCH_TYPE.PHOTO
                }

                R.id.user_search_radio_btn -> {
                    Log.d(TAG, "사용자검색 버튼 클릭!")
                    search_term_text_layout.hint = "사용자 검색"
                    search_term_text_layout.startIconDrawable =
                        resources.getDrawable(R.drawable.ic_person, resources.newTheme())
                    this.currentSearchType = SEARCH_TYPE.USER
                }
            }
        }


        // 텍스트가 변경이 되었을 때
        search_term_edit_text.onMyTextChanged {
            //입력된 글자가 하나라도 있으면 검색 버튼을 보여준다.
            if (it.toString().count() > 0) {
                frame_search_btn.visibility = View.VISIBLE
                search_term_text_layout.helperText = " "

                //스크롤뷰를 올린다.
                main_scrollview.scrollTo(0, 200)

            } else {
                frame_search_btn.visibility = View.INVISIBLE
            }

            if (it.toString().count() == 12) {
                Toast.makeText(this, "검색어는 12개까지만 입력가능합니다.", Toast.LENGTH_SHORT).show()
            }
        }

        //버튼 클릭시
        btn_search.setOnClickListener {
            Log.d(TAG, "검색 버튼 클릭됐음")
        }
    } //onCreate

    private fun handlesSearchButtonUI() {
        btn_progress.visibility = View.VISIBLE

        btn_search.text = " "

        Handler().postDelayed(
            {
                btn_progress.visibility = View.INVISIBLE
                btn_search.text = "검색"
            }, 1500
        )
    }
}